WooCommerce/Classes/ViewRelated/Coupons/Add and Edit Coupons/AddEditCouponViewModel.swift

```diff
@@ -0,0 +1,93 @@
+import Foundation
+import Yosemite
+
+/// View model for `AddEditCoupon` view
+///
+final class AddEditCouponViewModel: ObservableObject {
+
+    private let siteID: Int64
+    private let editingOption: EditingOption
+    private let discountType: Coupon.DiscountType
+    var titleView: String {
```

Comment:
Nit: since this is a string, maybe just naming it `title` is sufficient? `titleView` sounds more like a view to me.